web/modules/custom/dpl_library_agency/src/Form/GeneralSettingsForm.php

```diff
@@ -247,9 +252,37 @@ public function buildForm(array $form, FormStateInterface $form_state): array {
   }
 
   /**
+   * Validates interest_periods_config and default_interest_period_config.
+   *
+   * The interest_periods_config is validated for the format [days]-[label].
+   *
+   * The validation for default_interest_period_config is checking
+   * that the selected default value is present in interest_periods_config.
+   *
    * {@inheritdoc}
+   *
+   * @throws \Safe\Exceptions\PcreException
    */
   public function validateForm(array &$form, FormStateInterface $form_state): void {
+    $interestPeriods = $form_state->getValue('interest_periods_config');
+    $defaultInterestPeriod = $form_state->getValue('default_interest_period_config');
+
+    $interestPeriodsArray = explode(PHP_EOL, $interestPeriods);
+    foreach ($interestPeriodsArray as $period) {
+      if (!preg_match('/^[\d]+-[\w ]+$/m', trim($period))) {
```

Comment:
Don't know if it makes a difference, but is it intentionally that you wrote a whitespace after the word boundary character class?
`/^[\d]+-[\w ]+$/m`

Reply:
This is on purpose. If not, you are not able to do whitespaces. Example "1 month".